src/CalcManager/CEngine/scicomm.cpp

```diff
@@ -514,7 +514,7 @@ void CCalcEngine::ProcessCommandWorker(WPARAM wParam)
             m_HistoryCollector.PopLastOpndStart();
 
             m_bNoPrevEqu = true;
-        } while (m_precedenceOpCount >= 0);
+        } while (true);
```

Comment:
This would be the simple fix, but I think we could clean this up a bit to make things more readable:

1. Move the if/else from lines 468-499 into a new function, e.g. `void ResolveCurrentPrecedenceOperation()`
2. Convert the remaining code in the do/while loop into a regular while loop, using the condition at line 501: `while (m_precedenceOpCount > 0 && m_fPrecedence)`
3. Call the function from (1) at the bottom of the while loop.

```c++
ResolveCurrentPrecedenceOperation();

// Evaluate the precedence stack 
while (m_precedenceOpCount > 0 && m_fPrecedence)
{
    --m_precedenceOpCount;
    m_nOpCode = m_nPrecOp[--m_precedenceOpCount];
    m_lastVal = m_precedenceVals[m_precedenceOpCount];

    // Precedence Inversion check
    ni = NPrecedenceOfOp(m_nPrevOpCode);
    nx = NPrecedenceOfOp(m_nOpCode);

    if (ni <= nx)
    {
        m_HistoryCollector.EnclosePrecInversionBrackets();
    }
    m_HistoryCollector.PopLastOpndStart();

    m_bNoPrevEqu = true;

    ResolveCurrentPrecedenceOperation();
}
```

It isn't perfect, but I don't like having a `while(true)` with the condition buried in the center of the loop.

Reply:
Sure thing!

Reply:
Edited my code block above since I realized I had a double-decrement in the while loop.